Vagrantfile

```diff
@@ -90,16 +94,23 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     (1..2).each do |i|
       server.vm.provision "shell",
         inline: "echo \"Attempting to create tunnel 1\" && " \
-                "curl -s -X POST -d server=172.17.17.#{2*i} localhost:8080/ " \
+                "curl -s -X POST -d server=172.17.17.#{2*i} 192.168.69.100:8080/ " \
                 " > /dev/null 2>&1 || echo \"Error..?\""
       server.vm.provision "shell",
         run: "always",
-        inline: "curl -s -X POST localhost:8080/#{i}/"
+        inline: "curl -s -X POST 192.168.69.100:8080/#{i}/"
       server.vm.provision "shell",
         run: "always",
-        inline: "curl -s localhost:8080/#{i}/client_script/ " \
+        inline: "curl -s 192.168.69.100:8080/#{i}/client_script/ " \
                 "> /vagrant/tmp/proxy#{i}.sh"
     end
+    # change reverse path filtering to loose mode in order to
+    # allow incoming NATed traffic from the private networks
+    (1..2).each do |i|
+      server.vm.provision "shell",
+        run: "always",
+        inline: "echo '2' > /proc/sys/net/ipv4/conf/vpn-proxy-tun#{i}/rp_filter"
```

Comment:
This should be done by the application code every time a tunnel is started or stopped. Doing it in the vagrantfile will only work for this particular example with these two particular tunnels.